extend-expect.d.ts

```diff
@@ -9,7 +10,7 @@ declare global {
       toHaveProp(attr: string, value?: any): R;
       toHaveTextContent(text: string | RegExp, options?: { normalizeWhitespace: boolean }): R;
       toBeEnabled(): R;
-      toHaveStyle(style: object[] | object): R;
+      toHaveStyle(style: StyleProp<ViewStyle> | StyleProp<TextStyle>): R;
```

Comment:
1. Removed StypeProp: as it seems we want to allow only basic style objects, and not (nested) arrays.
2. Added `ImageStyle`

Reply:
I added the ImageStyle good catch. Regarding the styleProp, for the users it would be a breaking change as we won't support anymore this kind of tests https://github.com/testing-library/jest-native/blob/main/src/__tests__/to-have-style.js#L28.

If the idea is to remove the support, is fine for me just want to be sure it's intended.

Reply:
After #110 let's keep `StyleProp` support & add `ImageStyle`:

```ts
toHaveStyle(style: StyleProp<ViewStyle | TextStyle | ImageStyle>): R;
```

Reply:
This is a bit more permissive then `StyleProp<ViewStyle> | StyleProp<TextStyle> | StyleProp<ImageStyle>`